### PR TITLE
Feature/Expandable Design for Proprietary Data Exchange

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
SystemReques, OnSystemRequest, OnAppPermissionChanged, Enums, Structs index  
was updated with requestType = "OEM_SPECIFIC", requestSubType params.

### CODE REVIEW

**Comment by theresalech** [comment]
@KhrystynaDubovyk  you will need to complete the [CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform) in order for your pull request to be accepted.

**Comment by KhrystynaDubovyk** [comment]
@theresalech, I have successfully completed SDLC Contributor License Agreement on May, 22.
Please, let me know if any action is required from my side.
Many thanks.

File: docs/SDL/OnAppPermissionChanged/index.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ Purpose
 |appPermissionsConsentNeeded|Boolean|false||
 |appUnauthorized|Boolean|false||
 |priority|[Common.AppPriority](../../common/enums/#apppriority)|false||
-|requestType|[Common.RequestType](../../common/enums/#requesttype)|false|array: true<br>minsize: 0<br>maxsize: 100|
+|requestType|[Common.RequestType](../../common/enums/#requesttype)|false|array: true<br>minsize: 0<br>maxsize: 100|  
+|requestSubType|String|false|array: true<br> minsize: 0<br> maxsize: 100 <br> maxlength: 100|The list of SystemRequest's requestSubTypes allowed by policies for the named application.<br>If the app sends a requestSubType which is not specified in this list, then that request should be rejected.<br>An empty array signifies that any value of requestSubType is allowed for this app.<br> If this parameter is omitted, then a request with any value of requestSubType is now allowed for this app|
```
Comment (jacobkeeler) [inline docs/SDL/OnAppPermissionChanged/index.md]: There doesn't appear to be a 5th column in this table, so the description is ignored in this case

**Comment by jacobkeeler** [comment]
Also, merge conflicts should be fixed

File: docs/SDL/OnAppPermissionChanged/index.md (hunk 1 above)
Comment (KhrystynaDubovyk) [reply docs/SDL/OnAppPermissionChanged/index.md]: fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/98/commits/b882061b9308da7dbc4d0c60778d91b558bfb46c

**Comment by KhrystynaDubovyk** [comment]
@jacobkeeler, conflicts are fixed

File: docs/BasicCommunication/SystemRequest/index.md [hunk 2]
```diff
@@ -32,7 +35,8 @@ SDL validates all SystemRequests sent from the mobile app and returns DISALLOWED
 
 |Name|Type|Mandatory|Additional|
 |:---|:---|:--------|:---------|
-|requestType|[Common.RequestType](../../common/enums/#requesttype)|true||
+|requestType|[Common.RequestType](../../common/enums/#requesttype)|true||  
+|requestSubType|String|false|maxlength="255"|This parameter is filled for supporting OEM proprietary data exchanges.|
```
Comment (jacobkeeler) [inline docs/BasicCommunication/SystemRequest/index.md]: Missed this before, but it's the same case here, there is no 5th column

Comment (KhrystynaDubovyk) [reply docs/BasicCommunication/SystemRequest/index.md]: fixed https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/98/commits/ab7e9a293fa03fc5cae1ad15ff8809825e8196b3

**Pull request merged**